src/tests/Microsoft.PowerFx.Core.Tests/TexlTests.cs

```diff
@@ -4120,6 +4120,17 @@ public void TestBlobFunction(string expression, string expectedType)
             Assert.True(result.IsSuccess);
         }
 
+        [Fact]
+        public void TexlFunctionTypeSemanticTableConcat()
```

Comment:
>TexlFunctionTypeSemanticTableConcat [](http://example.com/codeflow?start=20&length=35)

This is a negative test - that's good. We should have positive tests as well (simple cases, type union, ...), that work as a spec for the behavior of this function